src/ScriptBuilder/CSharpCounterWriter.cs

```diff
@@ -6,7 +6,7 @@
 
     static class CSharpCounterWriter
     {
-        public static void WriteCode(string scriptPath, IEnumerable<TimerDefinition> timers, IEnumerable<MeterDefinition> meters, Dictionary<string, string> legacyInstanceNameMap)
+        public static void WriteCode(string scriptPath, IEnumerable<DurationDefinition> timers, IEnumerable<SignalDefinition> meters, Dictionary<string, string> legacyInstanceNameMap)
```

Comment:
Currently, `CSharpCounterWriter.WriteCode` uses `PerformanceCounterType.NumberOfItems32` for durations which according to doco.

> An instantaneous counter that shows the most recently observed value. Used, for example, to maintain a simple count of items or operations.

Is this the type of counter we want to use as we observe all the values now? Should we reconsider the type of this counter, or event, provide multiple ones, based on the same probe reading?